hosting/src/components/TogglePublishDocument.tsx

```diff
@@ -0,0 +1,37 @@
+import {Button} from "@/components/Button";
+import {useCrudTanamDocument, useTanamDocument} from "@/hooks/useTanamDocuments";
+import {serverTimestamp} from "firebase/firestore";
+import {useParams} from "next/navigation";
+
+export function TogglePublishDocument() {
+  const {documentId} = useParams<{documentId: string}>() ?? {};
+  const {data: document} = useTanamDocument(documentId);
+  const {update} = useCrudTanamDocument();
+
+  async function onTogglePublishDocument() {
+    let data = document?.toJson();
```

Comment:
**Do not downcast nice classes to JSON for logic.** 

JSON is only for serialization
 1. Saving data to database
 2. Transporting data to/from API endpoints

Keep the object as it is. Only cast it to JSON inside of hooks/services at the very moment when you are saving it to the database.